src/wrapper_array.jl

```diff
@@ -80,7 +80,8 @@ Base.axes(a::NamedDimsArray, d) = axes(parent(a), dim(a, d))
 
 function named_size(a::AbstractArray{T,N}) where {T,N}
     L = dimnames(a)
-    return NamedTuple{L,NTuple{N,Int}}(size(a))
+    axes_a = axes(a)
+    return NamedTuple{L,typeof(axes_a)}(axes_a)
 end
```

Comment:
I think `named_size` ought to return the size; there could be a `named_axes` which would return this.

Reply:
And (I can't comment on those lines) I'm not sure `similar(A)` and `similar(A, T)` should go via `named_size` at all, since it's not guaranteed to work:
```julia
julia> nda = NamedDimsArray(rand(2,2), (:x, :x));

julia> similar(nda)
ERROR: duplicate field name in NamedTuple: "x" is not unique
```
Probably they should just call `NamedDimsArray(similar(parent(A)), dimnames(A)`.

Reply:
This last method solves the original problem I was facing, so I'll revert `named_size` and get rid of `named_axes` entirely.